### ISSUE TITLE
Mod user directory not created or used when playing a mod

### ISSUE THREAD

**Issue by hemebond** [DarkPlacesEngine/darkplaces#150]
Using the latest master branch, mod content is loaded, but the profile directory is still `id1` which means config changes and save games all go there instead of `~/.darkplaces/zer`

```shell
$ ./darkplaces-sdl -game zer
Game is DarkPlaces-Quake using base gamedir id1
gamename for server filtering: DarkPlaces-Quake
Playing registered version.
DarkPlaces-Quake Linux, buildstring: v20140513-1441-g02cc3403 Mar 26 2024 release
```

**Comment by hemebond** [comment]
The commit to break it appears to be 5db369bea3d0cdd3dbd3b5b8038a399f0dbf84b6

**Comment by Baker7** [comment]
Quake compatibility is doing what Quake does.

If you start quake.exe -game zer, saves and configs are written to quake/zer folder.

This has always been the behavior of Quake and has always been the behavior of DarkPlaces.

**Comment by ladyhavoc** [comment]
I agree with Baker7 on this one, the behavior has always been that the fs_userdir is equal to the last entry in the search path, that is to say if you are using -game dpmod and do saveconfig test.cfg, it will save to ~/.darkplaces/dpmod/test.cfg and if you do exec test.cfg it will load ~/.darkplaces/dpmod/test.cfg, if there is no test.cfg in ~/.darkplaces/dpmod/ it will then try ./dpmod/test.cfg and then ~/.darkplaces/id1/test.cfg and then ./id1/test.cfg and if none of those are found it will print an error that it could not exec it because there is no file.

**Comment by bones-was-here** [comment]
@ladyhavoc yes, but should the `gamedir` command behave the same as the `-game` arguments? As in, should `gamedir` be able to change the primary gamedir (the one that is written to)? The old comments and code conflict about this.

If `gamedir` can change the primary gamedir, the following comment needs to be removed as there's really no protection: even if it's blocked while SVQC and/or CSQC are running, CSQC can replace the menu.dat or use localcmd("defer... to easily get around that.
```
The game directory can never be changed while quake is executing.  This is a
precaution against having a malicious server instruct clients to write files
over areas they shouldn't.
```

If protection is important then `gamedir` needs to be blocked from changing the primary gamedir at all times.

**Comment by Baker7** [comment]
The "gamedir" command originated in Quakeworld.

DarkPlaces supports connecting to and playing on Quakeworld servers.  When you connect to a Quakeworld server, the Quakeworld server will change your directory to whatever is required.

Typically it will change your directory to quake/qw, but it might change your game directory to "quake/fortress" or some other directory based on what the server is running.

Today, all Quake engines anyone uses has a "gamedir" command --- although the command called "game" in FitzQuake derivative engines (Quakespasm, Ironwail, VKQuake, etc.)

In every Quake/Quakeworld engine that has "gamedir" or "game" as a command, it functions the same as -game parameter.

**Comment by bones-was-here** [comment]
Ok, yes I noticed the gamedir change when joining QW, so even if `gamedir` command was protected, it could be circumvented by forcing a client to join a QW server that uses the target gamedir.

Sounds like protection isn't a feasible goal.
